src/strands/tools/mcp/mcp_client.py

```diff
@@ -638,8 +638,11 @@ def _background_task(self) -> None:
         """
         self._log_debug_with_thread("setting up background task event loop")
         self._background_thread_event_loop = asyncio.new_event_loop()
-        asyncio.set_event_loop(self._background_thread_event_loop)
-        self._background_thread_event_loop.run_until_complete(self._async_background_thread())
+        try:
+            asyncio.set_event_loop(self._background_thread_event_loop)
```

Comment:
looks like integ test is failing:

```
=========================== short test summary info ============================
FAILED tests_integ/mcp/test_mcp_client.py::test_streamable_http_mcp_client_with_500_error - AssertionError: Regex pattern did not match.
 Regex: 'Connection to the MCP server was closed'
 Input: 'Event loop is closed'
= 1 failed, 182 passed, 36 skipped, 1 xfailed, 85 warnings in 212.53s (0:03:32) =
```

Reply:
Ah. Moved closing the event loop until after the background thread has completed.